janus.c

```diff
@@ -217,6 +223,11 @@ static void janus_termination_handler(void) {
 	janus_pidfile_remove();
 	/* Close the logger */
 	janus_log_destroy();
+	/* If we're daemonizing, we send an error code to the parent */
+	if(daemonize) {
+		int code = 1;
+		write(pipefd[1], &code, sizeof(int));
```

Comment:
I'd surround the write with a do - while, which checks if the return code is == -1 and errno is EINTR.